nemo_aligner/algorithms/ppo.py

```diff
@@ -387,30 +507,36 @@ def fit(self):
                 step_metrics = {}
                 timing_metrics = {}
 
-                self.timer.start("rollout_time")
-                ppo_rollout_data, metrics = self.generate_rollouts(dataloader_iter, num_rollout_micro_batches)
-                self.timer.stop("rollout_time")
-                timing_metrics["rollout_time"] = self.timer.get("rollout_time")
+                critic_train_loop_amount = self.critic_warmup_steps if self.step == 0 else 1
 
-                # send critic train
-                clear_memory()
-                self.rm_critic.train(ppo_rollout_data)
+                for _ in range(critic_train_loop_amount):
```

Comment:
I have a small concern about this warmup mechanism:

This is going to consume some samples from the dataset (`self.consumed_samples` is increasing), but `self.step` doesn't increase as far as I can tell => this seems a bit fishy to me -- for instance I wonder if `max_steps` will still be valid?

In addition, this is to some extent "wasting" the early samples as we're not using them to update the actor.

What if instead we "reset" `consumed_samples` after the critic warmup, to start updating from sample 0? (another advantage is that we could evaluate the impact of this critic warmup without the confounding factor of not training the actor on the same data)

Reply:
hmmm yeah totally agree we should reset the consumed samples(and hopefully the first few steps will be more stable on the same data)

9bdc655ae0d08f1a92fb7e44e54cd3b130a44d83 <- i'm not 100% happy with the way I implemented this, but maybe this is good enough. the metrics are a little bit messed up(because we log consumed samples but then it gets reset)

Reply:
I see, it does seem a bit tricky to avoid it since we depend on `consumed_samples` increasing during critic warmup to go over the first samples. I guess it'd be possible to use a different counter but not sure it's worth the trouble, so I'm ok with it => resolving.